x-pack/plugins/canvas/public/state/middleware/aeroelastic.js

```diff
@@ -66,10 +66,16 @@ const elementToShape = (element, i) => {
   };
 };
 
-const updateGlobalPositions = (setMultiplePositions, { shapes, gestureEnd }, elems) => {
+const updateGlobalPositions = (setMultiplePositions, { shapes, gestureEnd }, unsortedElements) => {
+  const ascending = (a, b) => (a.id < b.id ? -1 : 1);
+  const relevant = s => s.type !== 'annotation' && s.subtype !== 'adHocGroup';
```

Comment:
These functions may be extracted in the `grouping` PR, although such invariant functions have no impact on performance here.